submission/submission_curriculum_track.py

```diff
@@ -0,0 +1,98 @@
+# pylint: disable=protected-access,bad-builtin,unused-variable,unused-import,ungrouped-imports
+
+'''Manual test for running a training epoch with curriculum'''
+#import wandb
+
+# train_helper provides the dummy functions/classes to make this run
+# TODO: replace these with the actual functions/classes
+from curriculum.train_helper import train_on_tasks, evaluate_on_tasks, load_agent_model
+from curriculum.train_helper import SyllabusTaskSampler
+
+# actual working code, with a lot to improve
+from curriculum.task_encoder import TaskEncoder
+from elm_for_nmmo.elm_curriculum_gen import SimpleTaskGenerator, OpenELMTaskGenerator
+
+# simulation participant's custom functions
+from submission import custom_curriculum as cc
+
+# if you want more task specs, use
+from curriculum import manual_curriculum as mc # and use mc.task_spec
+
+# Task spec, which is a list tuple and pickle-able, is passed around
+# task_spec: (reward_to, eval_fn, kwargs)
+# task_spec_with_embedding: (reward_to, eval_fn, kwargs, task_embedding)
+
+#####################################################################
+# this is the main training loop
+
+# LLM model path, which is shared by OpenELM and the task embedding generator
+LLM_CHECKPOINT = "Salesforce/codegen-350M-mono"
+
+# Joseph's comment: this will likely have to work off of a pretrained model
+AGENT_MODEL_PATH = ''
+
+NUM_TRAIN_TASKS = 30
+NUM_TEST_TASKS = 5
+NUM_NEW_TASKS = 5
+EPOCHS_PER_ELM_UPDATE = 10
+
+# setting DEBUG=True bypass running open elm
+DEBUG = True
+
+def train_with_curriculum(use_elm=True):
+  # participant's custom task spec is provided in a separate file, custom_curriculum.py
+  if use_elm:
+    # NOTE: passing cc.task_spec just to get some tasks out
+    # CHECK ME: would it be realistic to generate the training task on the fly here?
```

Comment:
Possibly we can generate templates such as Equip <item> <level> and fill them in with scripts?

Reply:
We may be able to use templates and/or prompts to "promote" diversity. What open elm does is basically prompting codegen model to generate code, and we can guide model by putting additional information in the prompt as code comments, like this: https://github.com/CarperAI/nmmo-baselines/blob/curriculum/elm_for_nmmo/nmmo_env.py#L161